lib/core/src/server/build-dev.ts

```diff
@@ -283,10 +283,9 @@ export async function buildDevStandalone(
 
     if (options.smokeTest) {
       await outputStats(previewStats, managerStats);
-      const managerWarnings = (managerStats as any).toJson().warnings.length > 0;
-      const previewWarnings =
-        !options.ignorePreview && (previewStats as any).toJson().warnings.length > 0;
-      process.exit(managerWarnings || previewWarnings ? 1 : 0);
+      const hasManagerWarnings = managerStats && (managerStats as any).toJson().warnings.length > 0;
```

Comment:
As types are now correct and match the objects you should be able to remove this `as any` and the one from next line.